### PR TITLE
Remove tox dependency

### PR DESCRIPTION
**Pull request by ssteinbach**
We've used `tox` for some time, and its served us well.  With the recent move to GitHub actions, however, we've found that it is no longer needed.  The reasons for removing it:

- one less build dependency
- the cmake/setup.py combination is already complicated, and tox is a further complication of that system
- Our makefile wasn't running `tox`, so this ensures that developers who use our developer makefile are matching _exactly_ what the ci does
- We'd like to do further things like run ci on more platforms and push wheels directly from ci into pypi/conda/etc -- this makes that more straightforward

What this PR does:

- moves the commands that `tox` was running into a makefile target, `ci-build-test`
- removes the setup.py and ci dependencies on `tox`
- switches the ci to use the new makefile target
- Removes references to and configuration for `tox` from various places (mostly the `setup.cfg`).

Todo:

- [x] move `lcov` execution to `Makefile`
- [x] remove `tox` section of `setup.cfg`
- [x] check block in `setup.py` to see if we can remove extra logic whose comment references tox

### CODE REVIEW

**Comment by meshula** [comment]
A big positive is that we can now see the build logs, so we can find where the build is failing :)

**Comment by ssteinbach** [comment]
Making sure that the RTD build isn't busted.  Lets wait for that to finish before landing this.

**Comment by ssteinbach** [comment]
Those look good too, if this looks good to folks, I think this is ready to land:
https://opentimelineio-ssteinbach.readthedocs.io/en/remove_tox/

File: setup.py
Comment (ssteinbach) [inline setup.py:177]: @meshula whoops, I think we meant to delete this code, right?

**Comment by ssteinbach** [comment]
Closes #772

File: setup.py
Comment (meshula) [reply setup.py:177]: yes, our decision was to leave it to logic in cmake

File: setup.py
Comment (meshula) [inline setup.py:187]: for example, here, we deleted this line in that expectation

**Review by meshula — COMMENTED** [review]
There's one chunk to remove, as you asked. The CMake side of things already has logic to disable coverage if lcov is not available on the platform (e.g. msvc)

**Review by meshula — APPROVED** [review]

File: README.md
Comment (meshula) [inline README.md:121]: grammar, delete "if"

**Review by meshula — APPROVED** [review]

**Pull request merged**
